cmd/libs/go2idl/openapi-gen/generators/openapi.go

```diff
@@ -0,0 +1,521 @@
+	if tagValues == nil {
+		return false
+	}
+	for _, val := range tagValues {
+		if val == value {
+			return true
+		}
+	}
+	return false
+}
+
+// NameSystems returns the name system used by the generators in this package.
+func NameSystems() namer.NameSystems {
+	return namer.NameSystems{
+		"raw": namer.NewRawNamer("", nil),
+	}
+}
+
+// DefaultNameSystem returns the default name system for ordering the types to be
+// processed by the generators in this package.
+func DefaultNameSystem() string {
+	return "raw"
+}
+
+func Packages(context *generator.Context, arguments *args.GeneratorArgs) generator.Packages {
+	boilerplate, err := arguments.LoadGoBoilerplate()
+	if err != nil {
+		glog.Fatalf("Failed loading boilerplate: %v", err)
+	}
+	inputs := sets.NewString(context.Inputs...)
+	header := append([]byte(fmt.Sprintf("// +build !%s\n\n", arguments.GeneratedBuildTag)), boilerplate...)
+	header = append(header, []byte(
+		`
+// This file was autogenerated by openapi-gen. Do not edit it manually!
+
+`)...)
+
+	targets := []*types.Type{}
+	for i := range inputs {
+		glog.V(5).Infof("considering pkg %q", i)
+		pkg, ok := context.Universe[i]
+		if !ok {
+			// If the input had no Go files, for example.
+			continue
+		}
+		for _, t := range pkg.Types {
+			if hasOpenAPITagValue(t.CommentLines, tagTargetType) {
+				glog.V(5).Infof("target type : %q", t)
+				targets = append(targets, t)
+			}
+		}
+	}
+	switch len(targets) {
+	case 0:
+		// If no target package found, that means the generated file in target package is up to date
+		// and build excluded the target package.
+		return generator.Packages{}
+	case 1:
+		pkg := context.Universe[targets[0].Name.Package]
+		return generator.Packages{&generator.DefaultPackage{
+			PackageName: strings.Split(filepath.Base(pkg.Path), ".")[0],
+			PackagePath: pkg.Path,
+			HeaderText:  header,
+			GeneratorFunc: func(c *generator.Context) (generators []generator.Generator) {
+				return []generator.Generator{NewOpenAPIGen(arguments.OutputFileBaseName, targets[0], context)}
+			},
+			FilterFunc: func(c *generator.Context, t *types.Type) bool {
+				// There is a conflict between this codegen and codecgen, we should avoid types generated for codecgen
+				if strings.HasPrefix(t.Name.Name, "codecSelfer") {
+					return false
+				}
+				pkg := context.Universe.Package(t.Name.Package)
+				if hasOpenAPITagValue(pkg.Comments, tagValueTrue) {
+					return !hasOpenAPITagValue(t.CommentLines, tagValueFalse)
+				}
+				if hasOpenAPITagValue(t.CommentLines, tagValueTrue) {
+					return true
+				}
+				return false
+			},
+		},
+		}
+	default:
+		glog.Fatalf("Duplicate target type found: %v", targets)
+	}
+	return generator.Packages{}
+}
+
+const (
+	specPackagePath          = "github.com/go-openapi/spec"
+	openAPICommonPackagePath = "k8s.io/kubernetes/cmd/libs/go2idl/openapi-gen/generators/common"
+)
+
+// openApiGen produces a file with auto-generated OpenAPI functions.
+type openAPIGen struct {
```

Comment:
This implements Generator right?
At other places, we ensure and document that by adding the following line:
```
var _ Generator := openAPIGen{}
```